### DISCUSSION TITLE
Montage "2 Wide" on Chrome doesn't account for vertical scroll bar

### DISCUSSION THREAD

**Discussion by Simpler1** [ZoneMinder/zoneminder#3749] [General]
In the desktop Chrome if there are are more than two monitors defined and the Montage view layout is set to "2 Wide", the width of the vertical scroll bar is not accounted for in the "width: 50%".  This causes the monitors to end up displaying only one wide.

The problem "corrects" itself if the Layout dropdown is changed to something else and then changed back to "2 Wide" because the scrollbar is already visible and is accounted for in the 50% calculation.

This problem doesn't occur in Firefox since the Firefox scrollbar width is much narrower.

The easy fix for this is to change the `width` value from 50% to 49% in the Positions column of the MontageLayouts table in the database.

`UPDATE MontageLayouts SET Positions='{ "default":{"float":"left", "width":"49%","left":"0px","right":"0px","top":"0px","bottom":"0px"} }' WHERE Id=2;`

"3 Wide", "4 Wide", & "5 Wide" layouts should probably have similar changes.

**Comment by connortechnology** [comment]
I've certainly seen this before and fixed it back and forth. However it does not happen to me currently. I find the 49% fix distasteful as well. it leaves weird white space.  

So I think we need to dig deeper into why this is happening for you, because it doesn't for me and I'd like to know why.

**Reply by Simpler1** [reply]
What screen resolutions have you tried?  I see the problem on both 1920x1080 and 1280x1024.  These are the only two that I've tried.
I'm running Cinnamon, but I'm not sure if that's related.
I have 3 monitors defined.

On my 1920x1080 display, when I refresh the page, the class="monitor" width is 950px (and it doesn't fit two wide), 
but if I change to _3 Wide_ and then back to _2 Wide_, the width is 924px (and it fits 2 wide).
What are your widths for these two scenarios.

**Comment by connortechnology** [comment]
I run kde, on a 1080p monitor, full width.  monitor class is 952px on initial load and after switching to 3 wide and back.. monitors class is 1920px please note that the monitors ul is display: flex.

Are you using a different set of css like dark mode?

**Reply by Simpler1** [reply]
I get similar (bad) results running base css or dark css.  

Dark:
Width of monitor class on initial load is **949px** and after switching to 3 wide and back, it's **942px** and fits.

Base:
Width of monitor class on initial load is **959px** and after switching to 3 wide and back, it's **952px** and fits.

Also, all three of my monitors are defined as 640x480 and the Montage settings for Width, Height, & Scale are all set to Auto.

Here's a snippet of what my scrollbar looks like in Chrome:
![image](https://github.com/ZoneMinder/zoneminder/assets/5922273/f55f754a-debf-49e1-ac57-fd77f0c24b62)
